### PR TITLE
add option to use amend when updating state to reduce size

### PR DESCRIPTION
**Pull request by rucciva**
related to #41. This PR add option to use `--amend` flag to simulate behavior similar to `git commit -a --amend` and `git push -f` to maintain only one commit per branch and thus reduce the size the repository compared to one commit per change on the state.

### CODE REVIEW

File: cmd/git_backend.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ var gitBackendCmd = &cobra.Command{
 		t, err := template.New(gitHTTPBackendConfigPath).Parse(`
 terraform {
 	backend "http" {
-		address = "{{ .protocol }}://localhost:{{ .port }}/?type=git&repository={{ .repository }}&ref={{ .ref }}&state={{ .state }}"
+		address = "{{ .protocol }}://localhost:{{ .port }}/?type=git&repository={{ .repository }}&ref={{ .ref }}{{ if eq .amend "true" }}&amend=true{{ end }}&state={{ .state }}"
```
Comment (dee-kryvenko) [inline cmd/git_backend.go:37]:
```suggestion
		address = "{{ .protocol }}://localhost:{{ .port }}/?type=git&repository={{ .repository }}&ref={{ .ref }}&amend={{ .amend }}&state={{ .state }}"
```

Nitpicking but I usually prefer any template to be as plain as possible for readability

**Review by dee-kryvenko — CHANGES_REQUESTED** [review]
Sorry for radio silence last week - life happened. This is looking great, and especial thanks for house keeping with Go and versions upgrade. Left a little nitpick suggestion for you and we can get this in. Thanks again!

File: cmd/git_backend.go (hunk 1 above)
Comment (rucciva) [reply cmd/git_backend.go:37]: Thanks @dee-kryvenko , i was going to do it that way too, but noticed it trigger backend change to existing terraform state. Is that okay?

Comment (rucciva) [reply cmd/git_backend.go]: anyway, i've updated the templates

Comment (dee-kryvenko) [reply cmd/git_backend.go]: Ugh.. that's a good catch! Didn't think of that one, that is true. Hate to ask you to revert it back in this case, although uglier but is better from a backward compatibility standpoint.

Comment (rucciva) [reply cmd/git_backend.go:37]: done

Comment (dee-kryvenko) [reply cmd/git_backend.go:37]: Thanks!

**Review by dee-kryvenko — APPROVED** [review]

**Pull request merged**